security

smriti548 opened the pull request:
updated the security and solved assigned issue AC-417

smriti548 committed:
Security

security classes and methods used for hashing passwords and encrypting data

smriti548 committed:
Merge pull request #1 from smriti548/smriti548-patch-1

Security

shivtej1505 commented on Security:
```diff
@@ -0,0 +1,93 @@
+The org.openmrs.util.Security class provides basic encryption and decryption methods for use in the API.
+The security class used by Andro client is :
+java.lang.Object
+org.openmrs.util.Security and then public class that has to be implemented or added extends Object class.
```
No need to go into so much depth, remember, this is a guide is for any user who wants to use Android Client and he/she may not have knowledge about coding.

smriti548 replied:
But until and unless a user is not aware about the class used under which those encryption and decryption methods are being used for password security how could he use those features

shivtej1505 replied:
You can just write that "we use _some algorithm_ which use _xxx bit key_ and fairly secure". You can also add 1-2 sentence about the working of the algorithm.

smriti548 replied:
what about these methods? should i remove all of them?

shivtej1505 commented on Security:
```diff
@@ -0,0 +1,93 @@
+Security
+OpenMRS is in contact with Security PS, an enterprise security consulting company.Security PS donated time to enhance the security of OpenMRS. Their application security experts found and privately reported software security weaknesses to the OpenMRS team. 
```
Could you point me to the link where you found this?
Also, such information shouldn't be on a guide. Don't you think?

smriti548 replied:
ok i'll change the content

shivtej1505 reviewed (changes requested):
Please squash your commit and change commit message. Check PR tips here: https://wiki.openmrs.org/display/docs/Pull+Request+Tips.

You didn't cover how the data exchanged with a server.

smriti548 committed:
updated details about the algorithm used 

updated details about the algorithm used and reduced some not much important methods, to make it more precious and easy to read.

smriti548 committed:
Update Security

updated details about the algorithm used and reduced some not much important methods, to make it more precious and easy to read.

smriti548 committed:
Merge pull request #2 from smriti548/smriti548-patch-1

Update Security

f4ww4z commented on Security:
```diff
@@ -0,0 +1,51 @@
+Security
```
Make this a header with `#`.

smriti548 replied:
sure will do it

f4ww4z reviewed (changes requested):
Please change the PR title to [AC-...] and follow the Pull Request tips as mentioned above.
https://wiki.openmrs.org/display/docs/Pull+Request+Tips

Can you also change the file to `security.md` (markdown file)?

shivtej1505 commented:
@smriti548 , are you still working on it?
